pina/geometry/simplex.py

```diff
@@ -0,0 +1,218 @@
+
+        # check consistency of simplex_dict
+        check_consistency(simplex_dict, dict)
+        for vertex in simplex_dict.values():
+            if not isinstance(vertex, list):
+                raise ValueError(f"{type(vertex).__name__} must be {list}.")
+
+        # vertices, vectors, dimension
+        self._vertices = simplex_dict
+        self._vert_list = torch.tensor(list(simplex_dict.values())).T
+        self._vectors = (
+            self._vertices_list[:, :-1] - self._vertices_list[:, None, -1]
+        ).type(torch.FloatTensor)
+
+        # build cartesian_bound
+        self._cartesian_bound = self._build_cartesian(
+            list(simplex_dict.values()), labels
+        )
+
+    @property
+    def variables(self):
+        """
+        Coordinate labels of simplex.
+
+        :return: Coordinate labels
+        :rtype: list[str]
+        """
+
+        return self._coordinate_labels
+
+    @property
+    def vertices(self):
+        """
+        Vertices of simplex.
+
+        :return: Vectors
+        :rtype: list[list]
+        """
+
+        return self._vertices
+
+    @property
+    def vectors(self):
+        """
+        Vectors.
+
+        :return: Vectors
+        :rtype: dict(LabelTensor)
+        """
+        return self._vectors
+
+    @property
+    def _vertices_list(self):
+        """
+        List of vectors.
+
+        :return: List of vectors
+        :rtype: list[LabelTensor]
+        """
+        return self._vert_list
+
+    @property
+    def cartesian_bound(self):
+        """
+        Cartesian border for Simplex domain.
+
+        :return: Cartesian border for Simplex domain
+        :rtype: CartesianDomain
+        """
+
+        return self._cartesian_bound
+
+    @property
+    def sample_surface(self):
+        """
+        Whether the surface should be sampled or not.
+
+        :return: Whether the surface should be sampled or not
+        :rtype: bool
+        """
+
+        return self._sample_surface
+
+    def _build_cartesian(self, vertices, labels):
+        """
+        Build Cartesian border for Simplex domain to be used in sampling.
+
+        :param vertices: list of Simplex domain's vertices
+        :type vertices: list[list]
+        :return: Cartesian border for triangular domain
+        :rtype: CartesianDomain
+        """
+
+        span_dict = {}
+
+        for i, coord in enumerate(labels):
+            sorted_vertices = sorted(vertices, key=lambda vertex: vertex[i])
+
+            # respective coord bounded by the lowest and highest values
+            span_dict[coord] = [sorted_vertices[0][i], sorted_vertices[-1][i]]
+
+        return CartesianDomain(span_dict)
+
+    def is_inside(self, point, check_border=False):
+        """
+        Check if a point is inside the simplex.
+        Uses the algorithm described here: 
+        https://math.stackexchange.com/questions/1226707/how-to-check-if-point-x-in-mathbbrn-is-in-a-n-simplex
```

Comment:
there are plenty many good references. Please put a book/ article on barycentric coordinates.

Reply:
Added a link to wikipedia article on barycentric coordinates.